packages/flutter/lib/src/cupertino/cupertino_focus_halo.dart

```diff
@@ -0,0 +1,105 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+import 'constants.dart';
+
+/// {@template flutter.cupertino.CupertinoFocusHalo}
+/// Applies a Cupertino-style focus border around its child when any of child focus nodes gain focus.
+///
```

Comment:
I think we can also add another paragraph here explaining a problem, which was brought up by @justinmc today, that the shape here does not automatically match the child widget inside, including the corner radius. Something like
```dart
  /// The shape of the focus halo does not automatically adapt to the child widget
  /// it encloses. You are responsible for specifying a shape that correctly
  /// matches the child's geometry by using the appropriate constructor, such as
  /// [CupertinoFocusHaloonRect] or [CupertinoFocusHalo.onRRect].
```